apps/bsd/src/screens/WriteInAdjudicationScreen.tsx

```diff
@@ -0,0 +1,473 @@
+}): JSX.Element => {
+  return React.createElement(
+    as,
+    { style: { display: 'flex', flex: '1', flexDirection, ...style } },
+    null,
+    ...(Array.isArray(children) ? children : [children])
+  )
+}
+
+const VStack = ({
+  as,
+  children,
+  style,
+}: {
+  as?: string
+  children: React.ReactNode
+  style?: React.CSSProperties
+}): JSX.Element => (
+  <Stack as={as} flexDirection="column" style={style}>
+    {children}
+  </Stack>
+)
+
+const HStack = ({
+  as,
+  children,
+  style,
+}: {
+  as?: string
+  children: React.ReactNode
+  style?: React.CSSProperties
+}): JSX.Element => (
+  <Stack as={as} flexDirection="row" style={style}>
+    {children}
+  </Stack>
+)
+
+const Spacer = ({
+  flex = 1,
+}: {
+  flex?: React.CSSProperties['flex']
+}): JSX.Element => <div style={{ flex }} />
+
+const Header = styled.div`
+  font-size: 3em;
+  font-weight: 900;
+`
+
+const ContestHeader = styled.div`
+  font-size: 1.5em;
+  font-weight: 900;
+`
+
+const MainChildColumns = styled.div`
+  flex: 1;
+  display: flex;
+  margin-bottom: -1rem;
+  > div {
+    margin-right: 1em;
+    &:first-child {
+      flex: 1;
+    }
+    &:last-child {
+      margin-right: 0;
+    }
+  }
+  button {
+    margin-top: 0.3rem;
+  }
+`
+
+const HIGHLIGHTER_COLOR = '#fbff0016'
+const FOCUS_COLOR = '#fbff007d'
+const EXTRA_WRITE_IN_MARGIN_PERCENTAGE = 0.2
+
+const WriteInAdjudicationBox = styled.div`
+  background-color: #ffffff;
+  padding: 20px 15px;
+  margin: 1em 0;
+`
+
+export type WriteInsByOption = Record<ContestOption['id'], string>
+export type WriteInsByContestAndOption = Record<Contest['id'], WriteInsByOption>
+
+const WriteInLabel = ({
+  contest,
+  writeIn,
+}: {
+  contest: CandidateContest
+  writeIn: WriteInAdjudicationReasonInfo
+}) => (
+  <HStack>
+    <div>
+      <strong>Write-In</strong>
+    </div>
+    <Text style={{ margin: '0 0.25em' }}>
+      (line {writeIn.optionIndex - contest.candidates.length + 1})
+    </Text>
+  </HStack>
+)
+
+const WriteInImage = ({
+  imageURL,
+  bounds,
+}: {
+  imageURL: string
+  bounds: Rect
+}) => (
+  <CroppedImage
+    src={imageURL}
+    alt="write-in area"
+    crop={{
+      x: bounds.x,
+      y: bounds.y - bounds.height * EXTRA_WRITE_IN_MARGIN_PERCENTAGE,
+      width: bounds.width,
+      height: bounds.height * (1 + 2 * EXTRA_WRITE_IN_MARGIN_PERCENTAGE),
+    }}
+    style={{
+      boxShadow: '4px 4px 3px 3px #999',
+      width: '50%',
+    }}
+  />
+)
+
+interface ContestAdjudicationProps {
+  imageURL: string
+  contest: CandidateContest
+  layout: BallotPageContestLayout
+  writeInsForContest: readonly WriteInAdjudicationReasonInfo[]
+  onInput?(optionId: ContestOption['id'], value: string): void
+  writeInValues?: WriteInsByOption
+}
+
+const ContestAdjudication = ({
+  imageURL,
+  contest,
+  layout,
+  writeInsForContest,
+  onInput,
+  writeInValues,
+}: ContestAdjudicationProps): JSX.Element => {
+  const onInputInternal: React.FormEventHandler = useCallback(
+    (event) => {
```

Comment:
nit: I think we can type event here which can probably let you avoid the as HTMLInputElement below but I'm not sure what the right type is.

Reply:
```ts
  const onInput: React.FormEventHandler = useCallback(
    (event) => {
      const input = event.currentTarget as HTMLInputElement
```